core/src/main/scala/org/apache/spark/deploy/history/HistoryPage.scala

```diff
@@ -46,6 +63,8 @@ private[history] class HistoryPage(parent: HistoryServer) extends WebUIPage("")
                 <script>setAppLimit({parent.maxApplications})</script>
             } else if (requestedIncomplete) {
               <h4>No incomplete applications found!</h4>
+            } else if (eventLogsUnderProcessCount > 0) {
+              <h4>No completed applications found!</h4>
```

Comment:
You may want to rebase your code and see if this is still the cleanest way of doing this. The code below this was recently changed